src/nvme/fabrics.c

```diff
@@ -650,10 +650,26 @@ int nvmf_add_ctrl(nvme_host_t h, nvme_ctrl_t c,
 	return nvme_init_ctrl(h, c, ret);
 }
 
+
 nvme_ctrl_t nvmf_connect_disc_entry(nvme_host_t h,
 				    struct nvmf_disc_log_entry *e,
 				    const struct nvme_fabrics_config *cfg,
 				    bool *discover)
+{
+	struct nvmf_connect_disc_entry2_args args = {
+		.args_size = sizeof(args),
+		.defcfg = cfg,
```

Comment:
I'm not happy with the pointer approach; that way we will always have two allocations when using nvmf_connect_disc_entry_2.
I'd rather go for either embed struct nvme_fabrics_config into nvmf_connect_disk_entry2 (and copy data over), or simply have two structures, passing both of them to nvmf_connect_disc_entry2()

Reply:
We can't avoid the allocation for `struct nvme_fabrics_config` as it already part of the current API. I'm fine with either idea. I have a slight preference for `nvme_fabrics_config` and `nvmf_connect_disk_entry2_args` as separate arguments.

Though first let's figure it out if we need it at all.